data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/prepper/peerforwarder/discovery/PeerListProviderFactory.java

```diff
@@ -36,10 +41,16 @@ public PeerListProvider createProvider(final PluginSetting pluginSetting) {
             case STATIC:
                 final List<String> endpoints = (List<String>) pluginSetting.getAttributeOrDefault(PeerForwarderConfig.STATIC_ENDPOINTS, null);
                 Objects.requireNonNull(endpoints, String.format("Missing '%s' configuration value", PeerForwarderConfig.STATIC_ENDPOINTS));
+                final List<String> invalidEndpoints = endpoints.stream().filter(endpoint -> !this.validateEndpoint(endpoint)).collect(Collectors.toList());
+                Preconditions.checkArgument(invalidEndpoints.size() == 0, "Including invalid endpoints: %s", invalidEndpoints);
```

Comment:
nit - Prefer `isEmpty()` here

Reply:
Also considering the use of [checkArgument](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkArgument(boolean,%20java.lang.Object)) vs [checkState](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkState(boolean,%20java.lang.Object))... The list of endpoints isn't exactly an argument being passed to the method, but it _is_ sort of passed in via the config file.

There's also the Apache Commons [Validate method](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/Validate.html#isTrue-boolean-), so many choices :P

Reply:
makes sense to use checkState